.github/workflows/wasm-js.yml

```diff
@@ -0,0 +1,33 @@
+name: WASM (JS)
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  msrv:
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-hack
+      - run: >-
+          cargo hack check --rust-version --workspace --all-targets
+          --ignore-private
+        working-directory: common
+
+  build:
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+      - name: Cargo Build
+        run: cargo build --verbose --lib --target wasm32-unknown-unknown --no-default-features --features wasm_js
+        working-directory: common
+      # TODO: Figure out a strategy for running unit tests; probably requires a web browser for time and randomness...
```

Comment:
I don't think this is a big deal initially; we can look at this later once we have wasm_bindgen + JS bindings working in a browser for some trivial case.

Reply:
Is integrating wasm-pack a good idea? wasm-pack can produce an npm package with a JS wrapper to the Rust core, and also has built-in headless browser unit test support built-in.

Reply:
Quite possibly. I'll take a look at your POC as well as wasm_pack docs before our Monday meeting and try to make an informed decision on this.

I also found a few test harness options that look really promising but many are not that stable yet so I'm fine with trying something first and changing later if necessary.

Reply:
Do you have any comments on this which need to be addressed _before_ this PR is merged @CatMe0w? I think your POC repo will necessarily have to build on the stuff in this branch, so I'd say let's get this merged unless you have any other blocking feedback.

Reply:
My current work is built on this branch right now and I think this PR in done and can be merged. Actually I am also waiting for this to be merged as I thought you might still have things to do on this branch. My bad, I should have said that earlier!

Reply:
No worries :)